profiling/src/interrupter.rs

```diff
@@ -0,0 +1,192 @@
+                let mut active = false;
+
+                // The goal is to wait on ticker if the associated PHP thread
+                // is serving a request. A tick or two after it ends may occur,
+                // but if the PHP thread is idle, like if it's being kept open
+                // with Connection: Keep-Alive, then we really need to limit
+                // how much we wake up.
+
+                // The code below has two select!s, where one of the recv's is
+                // the same -- the code to handle receiving messages. The
+                // difference is we only recv on the ticker if we're in an
+                // `active` state.
+                loop {
+                    if active {
+                        select! {
+                            // Remember to duplicate any changes to the else
+                            // branch below.
+                            recv(receiver) -> message => match message {
+                                Ok(Message::Start) => {
+                                    active = true;
+                                }
+                                Ok(Message::Stop) => {
+                                    active = false;
+                                }
+                                _ => {
+                                    break;
+                                }
+                            },
+
+                            recv(ticker) -> message => match message {
+                                Ok(_) => signal_pointers.notify(),
+
+                                // How does a timer fail?
+                                Err(err) => {
+                                    warn!("{err:#}");
+                                    break;
+                                }
+                            },
+                        }
+                    } else {
+                        // It's not great that we duplicate this code, but it
+                        // was the only way I could get it to compile.
+                        select! {
+                            recv(receiver) -> message => match message {
+                                Ok(Message::Start) => {
+                                    active = true;
+                                }
+                                Ok(Message::Stop) => {
+                                    active = false;
+                                }
+                                _ => {
+                                    break;
+                                }
+                            },
+                        }
+                    }
+                }
+
+                trace!("thread {thread_name} shut down");
+            });
+
+            Self {
+                sender,
+                join_handle: Some(join_handle),
+            }
+        }
+
+        pub fn start(&self) -> anyhow::Result<()> {
+            self.sender.send(Message::Start).map_err(|err| {
```

Comment:
Move this to send with timeout in case the other thread has crashed.

Reply:
The precaution of timeouts seem unnecessary as Florian and I could not get this to deadlock when causing the associated thread to panic in various places.